src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryAddress.java

```diff
@@ -142,4 +143,50 @@ static MemoryAddress ofLong(long value) {
                 NULL :
                 MemorySegmentImpl.NOTHING.baseAddress().addOffset(value);
     }
+
+    // The following methods can be used in conjunction with the java.foreign API.
+
+    /**
+     * Obtain the base object (if any) associated with this address. This can be used in conjunction with
+     * {@link #getUnsafeOffset(MemoryAddress)} in order to obtain a base/offset addressing coordinate pair
+     * to be used with methods like {@link sun.misc.Unsafe#getInt(Object, long)} and the likes.
+     * <p>
+     * This method is <em>unsafe</em>. It's use can result in putting the VM in a corrupt state when used incorrectly,
+     * and is provided solely to cover use-cases that can not otherwise be addressed safely. When used incorrectly, there
+     * are no guarantees made about the behaviour of the program. Particularly, incorrect use is not guaranteed to
```

Comment:
I'd replace `Particularly, incorrect use is not guaranteed to result in a VM crash, but might instead silently cause memory to be corrupted.` with `More specifically, incorrect uses of this method might result in a JVM crash or, worse, might silently result in memory corruption.`